lib/utils/references/resolveReference.js

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+function resolveReference(path, obj) {
+  let i,
+      ref = obj;
+
+  if (!Array.isArray(path)) {
+    return;
+  }
+
+  for (i = 0; i < path.length; i++) {
```

Comment:
now that it's been decided to move away from lodash, and the "utils" folder has started to contain lodash-like functions, do you think we should isolate this as `deepFind` function there?